packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -142,6 +176,23 @@ async def _call_in_executor(self, *args, **kwargs) -> Coroutine[Any, Any, str]:
         _call_with_args = functools.partial(self._call, *args, **kwargs)
         return await loop.run_in_executor(executor, _call_with_args)
 
+    def update_prompt_template(format: str, template: str):
+        """
+        Changes the class-level prompt template for a given format.
+        """
+        prompt_templates[format] = PromptTemplate.from_template(template)
+
+    def prompt_template(self, format) -> PromptTemplate:
```

Comment:
I would prefer `get_prompt_template` as the method name, since this is not an attribute or property.